Healthy/Classes/Utilities/Animatable.swift

```diff
@@ -0,0 +1,69 @@
+import UIKit
+
+/// A public protocol that contain two functions that will be affectively on button loading appearance
+/// Conform to AnyObject because this protocol will work only with classes such ex: UIButton, UIView....etc
+protocol Animatable: AnyObject {
+    
+    /// Start animating while waiting another task is done
+    /// - Parameter color: indicator color
+    func startAnimation(withColor color: UIColor)
+    
+    /// Stop animating when the task is finished
+    func stopAnimation()
+}
+
+/// Default implementation to UIButton loading indicator
+extension Animatable where Self: UIButton {
+    
+    ///A  computed property that returns the first subview that is an instance of UIActivityIndicatorView, if one exists.
+    private var activityIndicator: UIActivityIndicatorView? {
+        return subviews.first(where: { $0 is UIActivityIndicatorView }) as? UIActivityIndicatorView
+    }
+    
+    /// Start animating while waiting another task is done
+    /// We use an associated object to store the original button title when startLoading() is called.
+    /// - Parameter color: indicator color
+    func startAnimation(withColor color: UIColor = .black) {
+        let originalTitle = title(for: .normal)
+        isEnabled = false
+        setTitle("", for: .normal)
+        self.setupIndicatorLayout()
+        activityIndicator?.color = color
+        activityIndicator?.startAnimating()
+        objc_setAssociatedObject(self,
+                                 AssociatedKeys.originalTitle,
+                                 originalTitle,
+                                 .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
```

Comment:
```suggestion
        objc_setAssociatedObject(self,
                                 AssociatedKeys.originalTitle,
                                 originalTitle,
                                 .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
```

How to use it without associated object? 

```swift
private final class ButtonActivityIndicatorView: UIActivityIndicatorView {}
```

and get the first subview with this type?!

Reply:
We need to discuss & try this